pkg/rollout/rollout.go

```diff
@@ -0,0 +1,87 @@
+package rollout
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	log = logf.Log.WithName("rollout")
+)
+
+// Rollout is abstract class to secure that every node updates it's state one by one
+type Rollout struct {
+	leadershipElector     *leaderelection.LeaderElector
+	startedLeadingChannel chan interface{}
+}
+
+// NewRollout creates new Rollou
+func NewRollout(cfg *rest.Config, scheme *runtime.Scheme) (*Rollout, error) {
+	startedLeadingChannel := make(chan interface{})
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize apiserver client: %v", err)
+	}
+
+	recorderProvider, err := NewProvider(cfg, scheme, log.WithName("record_provider"))
```

Comment:
Why we don't pass directly clientset to NewRollout and NewProvider ?

Reply:
NewProvider is a copy paste, but you are right, we could make it so it takes the clientSet directly. @HlinaCZ could you adjust our copied module, so it takes the clientset we create here instead of cnv and scheme?

For NewRollout, the reason is, we don't need client set in layers above, only here, so I think it is better to just pass what we have and let NewRollout create the clientset. We can later think if we can get rid of it completely.

Reply:
We can pass clientset directly but we still have to pass scheme because `NewProvider` uses scheme for func `GetEventRecorderFor()`.